### PR TITLE
[flutter_tools] let the logger know about machine mode

### PR DESCRIPTION
**Pull request by jonahwilliams**
In order to better support machine mode, the loggers need to be aware of it - so that they can allow calling code to elide prompts or avoid displaying messages that aren't reasonable to JSON format.

Start of the work towards https://github.com/flutter/flutter/issues/80993

### CODE REVIEW

**Review by christopherfujino — APPROVED** [review]
cooLGTM

**Comment by jonahwilliams** [comment]
FYI @Hixie on linux testing failure: 

```
+ web_socket_channel 2.1.0
+ webkit_inspection_protocol 1.0.0
+ yaml 3.1.0
Changed 54 dependencies!
Cloning into '../../bin/cache/pkg/tests'...
Downloading Dart SDK from Flutter engine 1c24b5050df26a5ba80810b41d66dbcd1506e7f7...
Building flutter tool...
error: unknown option `:--:--'
usage: git checkout [<options>] <branch>
   or: git checkout [<options>] [<branch>] -- <file>...

```

**Comment by christopherfujino** [comment]
> FYI @Hixie on linux testing failure:
> 
> ```
> + web_socket_channel 2.1.0
> + webkit_inspection_protocol 1.0.0
> + yaml 3.1.0
> Changed 54 dependencies!
> Cloning into '../../bin/cache/pkg/tests'...
> Downloading Dart SDK from Flutter engine 1c24b5050df26a5ba80810b41d66dbcd1506e7f7...
> Building flutter tool...
> error: unknown option `:--:--'
> usage: git checkout [<options>] <branch>
>    or: git checkout [<options>] [<branch>] -- <file>...
> ```

I'll open an issue, since I also saw this.

File: packages/flutter_tools/lib/executable.dart [hunk 1]
```diff
@@ -77,8 +77,9 @@ Future<void> main(List<String> args) async {
   final bool muteCommandLogging = (help || doctor) && !veryVerbose;
   final bool verboseHelp = help && verbose;
   final bool daemon = args.contains('daemon');
-  final bool runMachine = (args.contains('--machine') && args.contains('run')) ||
-                          (args.contains('--machine') && args.contains('attach'));
+  final bool machine = args.contains('--machine');
+  final bool runMachine = (machine && args.contains('run')) ||
+                          (machine && args.contains('attach'));
```
Comment (Hixie) [inline packages/flutter_tools/lib/executable.dart:82]: this won't work. Consider for example "flutter create --machine --org run attach".

Comment (christopherfujino) [reply packages/flutter_tools/lib/executable.dart:82]: This is true. Also, this PR doesn't change this behavior.

Comment (Hixie) [reply packages/flutter_tools/lib/executable.dart:82]: Ah right this is what I filed in https://github.com/flutter/flutter/issues/75876

**Pull request merged**